### ISSUE TITLE
AKKA streams RetryFlow withBackoff not stopping after maxRetries

### ISSUE THREAD

**Issue by saftacatalinmihai** [akka/akka#28425]
The **RetryFlow.withBackoff** wrapper does not stop retrying after the maxRetries is reached.
It just goes on forever

Snippet:
https://scastie.scala-lang.org/nYS6qVs0SXuRFV9T2ZZJrQ

Inline:
```
import akka.NotUsed
import akka.actor.ActorSystem
import akka.stream.scaladsl.{ Flow, RetryFlow, Sink, Source }

import scala.concurrent.ExecutionContext
import scala.concurrent.duration._

object StreamSB extends App {

  implicit val as: ActorSystem      = ActorSystem()
  implicit val ec: ExecutionContext = as.dispatcher

  private val source: Source[Int, NotUsed] = Source(1 to 10)

  private val flow = Flow[Int].map { _ =>
    -1
  }

  val retryFlow = RetryFlow
    .withBackoff(
      minBackoff = 10.millis,
      maxBackoff = 100.millis,
      randomFactor = 0d,
      maxRetries = 3,
      flow
    )(
      decideRetry = {
        case (x, result) if result < 0 =>
          println("Neg result")
          Some(x)
        case (x, result) =>
          println("Pos result")
          None
      }
    )

  source.via(retryFlow).runWith(Sink foreach println)
}
```

**Comment by johanandren** [comment]
Looks like there's a missing counter increase when deciding to retry. Would you be up for PR:ing a fix?

**Comment by saftacatalinmihai** [comment]
I can try, but I haven't contributed to this codebase before... may take a while to figure it out.

**Comment by johanandren** [comment]
You can find some instructions in [CONTRIBUTING.md](https://github.com/akka/akka/blob/master/CONTRIBUTING.md) and if you get stuck feel free to join the `akka/dev` gitter channel and we'll try to help you out.

**Comment by saftacatalinmihai** [comment]
@johanandren please check created PR

**Comment by ennru** [comment]
Fixed with #28443
